immuni_exposure_ingestion/models/validators.py

```diff
@@ -7,58 +9,152 @@
         if len(rolling_start_numbers) != len(set(rolling_start_numbers)):
             raise ValidationError("Rolling start numbers are not unique.")
 
         next_rolling_start_number = sorted_teks[0].rolling_start_number
         for current in sorted_teks:
             if current.rolling_start_number < next_rolling_start_number:
                 raise ValidationError("Overlapping rolling start numbers.")
-            next_rolling_start_number = current.rolling_start_number + current.rolling_period
+            next_rolling_start_number = (
+                current.rolling_start_number + current.rolling_period
+            )
 
         if config.ALLOW_NON_CONSECUTIVE_TEKS:
             return
 
+        # TODO: Change this logic once ignored_teks are not ignored anymore.
         initial_start_number = rolling_start_numbers[0]
         if rolling_start_numbers != [
             initial_start_number + 144 * i for i in range(len(rolling_start_numbers))
         ]:
             raise ValidationError("Some TemporaryExposureKeys are missing.")
+
+    @staticmethod
+    def _today_rolling_start_number() -> int:
+        return int(
+            datetime.utcnow()
+            .replace(hour=0, minute=0, second=0, microsecond=0, tzinfo=timezone.utc)
+            .timestamp()
+            / timedelta(minutes=10).total_seconds()
+        )
+
+    def _filter_valid_teks(
+        self, teks: List[TemporaryExposureKey]
+    ) -> List[TemporaryExposureKey]:
+        """
+        From all given TEKs, return the list of valid ones.
+        Some may be valid for the final framework, but currently ignored.
+        Raise as soon as an invalid TEKs is found.
+
+        :param teks: the list of TEKs to be filtered.
+        :raises: ValidationError in case at least one TEK is deemed invalid.
+        :return: the list of valid TEKs.
+        """
+
+        valid_teks = ignored_teks = today_teks = list()
+        today_rolling_start_number = self._today_rolling_start_number()
+
+        for tek in teks:
+
+            if tek.rolling_period == _ROLLING_PERIOD_MAX:
+                valid_teks.append(tek)
+            elif not _ROLLING_PERIOD_MIN <= tek.rolling_period < _ROLLING_PERIOD_MAX:
+                # The TEKs is not coming from the exposure notification framework: immediate short
+                # circuit.
+                raise ValidationError(
+                    f"Some rolling_period values are not in "
+                    f"[{_ROLLING_PERIOD_MIN},{_ROLLING_PERIOD_MAX}] (e.g., {tek.rolling_period})."
+                )
+            else:
+                # Not to break the clients when the underlying framework (may) not support it yet,
+                # TEKs with rolling_period < 144 are ignored.
+                ignored_teks.append(tek)
+
+            if tek.rolling_start_number == today_rolling_start_number:
+                today_teks.append(tek)
+            elif tek.rolling_start_number > today_rolling_start_number:
```

Comment:
if my comment above is true, then this would fail when generating more than 1 tek in the current day :)